qa/tasks/cbt.py

```diff
@@ -0,0 +1,134 @@
+import logging
+import os
+import yaml
+
+from teuthology import misc
+from teuthology.config import config as teuth_config
+from teuthology.orchestra import run
+from teuthology.task import Task
+
+log = logging.getLogger(__name__)
+
+
+class CBT(Task):
+    """
+    Passes through a CBT configuration yaml fragment.
+    """
+    def __init__(self, ctx, config):
+        super(CBT, self).__init__(ctx, config)
+        self.log = log
+
+    def hosts_of_type(self, type_):
+        return [r.name for r in self.ctx.cluster.only(misc.is_type(type_)).remotes.keys()]
+
+    def generate_cbt_config(self):
+        mon_hosts = self.hosts_of_type('mon')
+        osd_hosts = self.hosts_of_type('osd')
+        client_hosts = self.hosts_of_type('client')
+        cluster_config = dict(
+            user=self.config.get('cluster', {}).get('user', 'ubuntu'),
+            head=mon_hosts[0],
+            osds=osd_hosts,
+            mons=mon_hosts,
+            clients=client_hosts,
+            osds_per_node=self.config.get('cluster', {}).get('osds_per_node', 1),
+            rebuild_every_test=False,
+            use_existing=True,
+            iterations=self.config.get('cluster', {}).get('iterations', 1),
+            tmp_dir='/tmp/cbt',
+            pool_profiles=self.config.get('cluster', {}).get('pool_profiles'),
+            )
+        return dict(
+            cluster=cluster_config,
+            benchmarks=self.config.get('benchmarks'),
+            )
+
+    def install_dependencies(self):
+        system_type = misc.get_system_type(self.first_mon)
+
+        if system_type == 'rpm':
+            install_cmd = ['sudo', 'yum', '-y', 'install']
+            cbt_depends = ['python-yaml', 'python-lxml', 'librbd-devel', 'pdsh']
+        else:
+            install_cmd = ['sudo', 'apt-get', '-y', '--force-yes', 'install']
+            cbt_depends = ['python-yaml', 'python-lxml', 'librbd-dev']
+        self.first_mon.run(args=install_cmd + cbt_depends)
+         
+        # install fio
+        testdir = misc.get_testdir(self.ctx)
+        self.first_mon.run(
+            args=[
+                'git', 'clone', '-b', 'master',
+                'https://github.com/axboe/fio.git',
+                '{tdir}/fio'.format(tdir=testdir)
+            ]
+        )
+        self.first_mon.run(
+            args=[
+                'cd', os.path.join(testdir, 'fio'), run.Raw('&&'),
+                './configure', run.Raw('&&'),
+                'make', run.Raw('&&'),
+                'sudo', 'make', 'install'
```

Comment:
should avoid 'make install' into a persistent dir, since the machines will get reused, and could get problems from different versions later

configure with --prefix=testdir/fio/build or something, so that it's cleaned up at the end?